### ISSUE TITLE
i:Restrict Content Pro: referrals created show incorrect Description and Reference values

### ISSUE THREAD

**Issue by gspice** [AffiliateWP/AffiliateWP#2389]
Reported on this HS ticket:
https://secure.helpscout.net/conversation/434509391/66990/?folderId=1170842

Reproduced on our sandbox:  http://s14890.p27.sites.pressdns.com/rcp/

* Use an affiliate link to register for a free level (Free Access GSC), then use the same browser, go to /register and purchase  paid level (Paid Level GSC).  The **Description** and **Reference** fields on the referral point to the free level entry in RCP payments.   Should be the paid level's description and payment reference #.

Debug log showed this on the above:
```
2017-9-20 20:34:19 - Visit #7 recorded for affiliate #4 in track_visit()
2017-9-20 20:35:38 - Referral could not be retrieved during complete_referral(). Value given: 2b4c9f47bdbf614d572e663d21983720
2017-9-20 20:37:29 - Pending Referral #86 created successfully.
2017-9-20 20:37:32 - Referral could not be retrieved during complete_referral(). Value given: 184e64dc5992360b700bd1fb9ea031f4
```
* Use an affiliate link to access the sandbox , go to /register and purchase the paid access level straight away (Paid Level GSC).  The **Description** and **Reference** fields will be blank on the referral. 

Debug log showed this on the above:
```
2017-9-20 20:41:55 - Visit #8 recorded for affiliate #4 in track_visit()
2017-9-20 20:43:12 - Pending Referral #87 created successfully.
2017-9-20 20:43:15 - Referral could not be retrieved during complete_referral(). Value given: 221e659358372059b67db20d61c78dc5
```

**Comment by mindctrl** [comment]
PR: #2392 

There were two issues here. RCP wasn't adding the pending key before the hook. We've fixed that and will release it today.

Another issue was that AffWP never introduced support for the pending subscription name, which was introduced at the same time pending key was. This resulted in the wrong referral description being shown when a member upgraded from one subscription to another. That's what is addressed in this PR. I also removed some of the duplicated logic for getting the pending key.

**Comment by pippinsplugins** [comment]
@gspice could you please test this PR? If all is well, I'll push out the update today.

**Comment by gspice** [comment]
Tested with both updated plugins and results were successful:
* Free level registration, then upgrade - on the referral, the **Description** was correct and the **Reference** link takes you to the correct payment record in RCP.
* Buying the full price membership outright was no longer blank on the **Description** and **Reference** fields on the referral.

**Comment by michaelbeil** [comment]
This issue has come up again.

**Comment by amdrew** [comment]
@michaelbeil Can you list your plugin versions and exact steps to replicate?

**Comment by michaelbeil** [comment]
I was using AffiliateWP v2.1.6 and RCP v2.9.1, purchased a free subscription through RCP through an affiliate link, and this error came up.

http://beil.io/rLeCSl

**Comment by mindctrl** [comment]
@michaelbeil that would be why you saw it. There was a fix in RCP 2.9.4 that went along with the fix here.

**Comment by michaelbeil** [comment]
Oh no. You are right. Apparently my license key got lost somehow. So, I updated to the latest version and couldn't reproduce the issue.
